Mathlib/Data/Fin/Restrict.lean

```diff
@@ -0,0 +1,138 @@
+/-
+Copyright (c) 2025 Wrenna Robson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Wrenna Robson
+-/
+import Mathlib.Data.Fin.Basic
+
+/-!
+# TBD
+
+## Main definitions
+
+-/
```

Comment:
Please don't increase the scope of a PR like this when it's almost ready to merge.